detectindent.lua

```diff
@@ -5,19 +5,50 @@ local util = import("micro/util")
 
 function onBufferOpen(buf)
     local spaces, tabs = 0, 0
+    local space_count, prev_space_count = -1, -1
```

Comment:
Perhaps the initial value for prev_space_count should be 0? The value -1 causes ignoring indentation in the first line of file.

Also, for that matter, maybe for more clarity we can avoid using -1 values and instead move the tabsizes logic inside `if r == " "` condition.

Reply:
Good idea, done. It does read better and keeps the bulk of the space indent code in one place.